var/spack/repos/builtin/packages/octave/package.py

```diff
@@ -182,12 +184,19 @@ def configure_args(self):
                 "--with-java-includedir=%s" % spec['jdk'].prefix.include,
                 "--with-java-libdir=%s"     % spec['jdk'].prefix.lib
             ])
+        else:
+            config_args.append("--disable-java")
 
         if '~opengl' in spec:
             config_args.extend([
                 "--without-opengl",
                 "--without-framework-opengl"
             ])
+        else:
+            config_args.extend([
+                "--without-opengl",
+                "--without-framework-opengl"
```

Comment:
Good point. Maybe comment the whole variant and if-statement out? But I won't hold you to it since this wasn't part of your PR to begin with. Approved!

P.S. Do you happen to know the difference between `opencl`, `opencv`, and `opengl`? Just curious. I need to add an `opencl` package next week for a user.